zebra-consensus/src/transaction/tests.rs

```diff
@@ -660,3 +660,24 @@ fn mock_sprout_join_split_data() -> (JoinSplitData<Groth16Proof>, ed25519::Signi
 
     (joinsplit_data, signing_key)
 }
+
+#[test]
+fn empty_sprout_pool_after_nu() {
+    zebra_test::init();
+
+    // get a block that we know it haves a transaction with `vpub_old` field greater than 0.
+    let block: Arc<_> = zebra_chain::block::Block::zcash_deserialize(
+        &zebra_test::vectors::BLOCK_MAINNET_419199_BYTES[..],
+    )
+    .unwrap()
+    .into();
+
+    // we know the 4th transaction of this block has the`vpub_old` field greater than 0.
+    assert_eq!(
+        check::disabled_sprout_pool(&block.transactions[3]),
+        Err(TransactionError::DisabledSproutPool)
+    );
+
+    // we know the in the 1st transaction of the same block, the `vpub_old` field is 0.
+    assert_eq!(check::disabled_sprout_pool(&block.transactions[0]), Ok(()));
```

Comment:
Test coverage: There are two different cases here, please check both of them:
- a transaction with no sprout joinsplits
- a transaction with some sprout joinsplits, but all their `vpub_old` fields are 0

This assertion checks the coinbase transaction, which is not allowed to have any sprout joinsplits.

Reply:
I made the change to test something else than the coinbase in https://github.com/ZcashFoundation/zebra/pull/2399/commits/9861b53b84367933ee7e9c888c3fc12052e5f5b1